src/spark_manager.py

```diff
@@ -347,33 +323,182 @@ def _create_master_service(self):
+    ) -> None:
+        """
+        Create a Kubernetes service, replacing it if it already exists.
+
+        Args:
+            service: The service definition
+            name: Name of the service
+            resource_description: Description of the resource for logging
+        """
         try:
             # Try to create the service first
             self.core_api.create_namespaced_service(
                 namespace=self.namespace, body=service
             )
-            logger.info(f"Created Spark master service: {self.master_name}")
+            logger.info(f"Created {resource_description}: {name}")
         except ApiException as e:
             if e.status == 409:  # Conflict - already exists
                 try:
                     # Delete the existing service
                     self.core_api.delete_namespaced_service(
-                        name=self.master_name, namespace=self.namespace
-                    )
-                    logger.info(
-                        f"Deleted existing Spark master service: {self.master_name}"
+                        name=name, namespace=self.namespace
                     )
+                    logger.info(f"Deleted existing {resource_description}: {name}")
 
                     # Create new service
                     self.core_api.create_namespaced_service(
                         namespace=self.namespace, body=service
                     )
-                    logger.info(f"Recreated Spark master service: {self.master_name}")
+                    logger.info(f"Recreated {resource_description}: {name}")
+                except ApiException as delete_error:
+                    logger.error(
+                        f"Error replacing {resource_description}: {delete_error}"
+                    )
+                    raise
+            else:
+                logger.error(f"Error creating {resource_description}: {e}")
+                raise
+
+    def _create_or_replace_deployment(
+        self, deployment: Dict, name: str, resource_description: str
+    ) -> None:
+        """
+        Create a Kubernetes deployment, replacing it if it already exists.
+
+        Args:
+            deployment: The deployment definition
+            name: Name of the deployment
+            resource_description: Description of the resource for logging
+        """
+        try:
+            # Try to create the deployment first
+            self.apps_api.create_namespaced_deployment(
+                namespace=self.namespace, body=deployment
+            )
+            logger.info(f"Created {resource_description}: {name}")
+        except ApiException as e:
+            if e.status == 409:  # Conflict - already exists
+                try:
+                    # Delete the existing deployment
+                    self.apps_api.delete_namespaced_deployment(
+                        name=name, namespace=self.namespace
+                    )
```

Comment:
Was going to say this deletion stuff makes me nervous but then remembered https://kbase-jira.atlassian.net/browse/CDM-107